File: sesion-5/secure-snippets-js-csharp/csharp/1-SqlInjection.cs
// ✅ SEGURO: Uso de parámetros SqlParameter
using System;
using System.Data.SqlClient;

class SqlSafeDemo {
    static void Main() {
        Console.Write("Usuario: ");
        var user = Console.ReadLine();

        using var conn = new SqlConnection("Server=.;Database=Demo;Trusted_Connection=True;");
        conn.Open();
        var sql = "SELECT * FROM Users WHERE Username = @username";
        var cmd = new SqlCommand(sql, conn);
        cmd.Parameters.AddWithValue("@username", user);
        var reader = cmd.ExecuteReader();
        while (reader.Read()) {
            Console.WriteLine(reader["Username"]);
        }
    }
}

File: sesion-5/secure-snippets-js-csharp/js/1-sql-injection.js
/**
 * ✅ SEGURO: Uso de parámetros preparados con mysql2
 */
const express = require('express');
const app = express();
const mysql = require('mysql2');

const conn = mysql.createConnection({ host: 'localhost', user: 'root', password: 'root', database: 'demo' });

app.get('/user', (req, res) => {
  const username = req.query.username;
  const sql = "SELECT * FROM users WHERE username = ?";
  conn.execute(sql, [username], (err, rows) => {
    if (err) return res.status(500).send('Error DB');
    res.json(rows);
  });
});

app.listen(3000, () => console.log('Servidor seguro en http://localhost:3000'));
